R/utility.R

```diff
@@ -725,4 +724,117 @@ viewSplines <- function(knots, degree, theta) {
   }
 
   return(p)
-}
\ No newline at end of file
+}
+
+#' Get survival curve parameters
+#'
+#' @param points A list of two-element vectors specifying the desired time and 
+#' probability pairs that define the desired survival curve
+#' @return A vector of parameters that define the survival curve optimized for
+#' the target points. The first element of the vector represents the "f"
+#' parameter and the second element represents the "shape" parameter.
+#' @examples
+#' points <- list(c(60, 0.90), c(100, .75), c(200, .25), c(250, .10))
+#' survGetParams(points)
+#' @export
+#' @concept utility
+survGetParams <- function(points) {
+  
+  assertNotMissing(points = missing(points))
+  assertClass(points = points, class = "list")
+  
+  time <- sapply(points, function(x) x[1])
+  p <- sapply(points, function(x) x[2])
+  
+  assertPositive(time)
+  assertAscending(time)
+  
+  assertProbability(p)  
+  assertDescending(p)
+  
+  loss_surv <- function(params, points) {
+    
+    loss <- function(a, params) {
+      ( params[2]*(log(-log(a[2])) - params[1]) - log(a[1]) ) ^ 2
+    }
+    
+    sum(sapply(points, function(a) loss(a, params)))
+    
+  }
+  
+  r <- stats::optim(
+    par = c(1, 1), 
+    fn = loss_surv, 
+    points = points,
+    method = "L-BFGS-B", 
+    lower = c(-Inf, 0),
+    upper = c(Inf, Inf)
+  )
+  
+  if (r$convergence !=0) stop("Optimization did not converge")
+  
+  return(r$par)
+}
+
+#' Plot survival curves
+#' 
+#' @param f This is the "formula" parameter of the Weibull-based survival curve
```

Comment:
Here maybe `weibull_formula` or `formula`  would be a bit  more descriptive (unless ofc the parameter is actually called f in the underlying distribution).

Reply:
Changed to `formula`